tests/unittests/BackendTest.cpp

```diff
@@ -165,6 +165,82 @@ TEST(RuntimeBundle, BundleSymbolInfo) {
   EXPECT_EQ(table.find("tensorview_reshape")->second.output, false);
 }
 
+// Test if the placeholders are allocated contiguously as
+// Input|InputOutput|Output.
+TEST(RuntimeBundle, ContiguousPlaceholder) {
+  ExecutionEngine EE;
+  PlaceholderBindings bindings;
+  auto &mod = EE.getModule();
+  Function *F = mod.createFunction("main");
+  Tensor inputs(ElemKind::FloatTy, {1, 4});
+  inputs.getHandle() = {1, 1.2f, 0.5f, 1.3f};
+
+  auto *A = mod.createPlaceholder(ElemKind::FloatTy, {1, 4}, "A", false);
+  auto *B = mod.createPlaceholder(ElemKind::FloatTy, {1, 4}, "B", false);
+  auto *Ex = mod.createPlaceholder(ElemKind::FloatTy, {1, 4}, "E", false);
+  auto *add = F->createAdd("add", A, Ex);
+  auto *sub = F->createSub("sub", B, add);
+  F->createSave("ret", sub);
+
+  LoweredInfoMap loweredMap;
+  CompilationContext cctx{&bindings, &loweredMap};
+  cctx.precisionConfig.quantMode = QuantizationMode::Profile;
+
+  bindings.allocate(A);
+  bindings.allocate(Ex);
+  EE.compile(F, cctx);
+
+  auto &table = EE.getCompiledFunction().getRuntimeBundle().getSymbolTable();
+
+  std::vector<glow::runtime::RuntimeSymbolInfo> tableContainer;
+  // Only check placeholders.
+  for (auto v : table) {
+    if (v.second.symbolCategory == glow::runtime::SymbolCategory::Placeholder) {
+      tableContainer.push_back(v.second);
+    }
+  }
+  // Sort the placeholders by offset.
+  sort(tableContainer.begin(), tableContainer.end(),
+       [](const glow::runtime::RuntimeSymbolInfo &a,
+          const glow::runtime::RuntimeSymbolInfo &b) {
+         return (a.offset < b.offset);
+       });
+
+  // Define the order of placeholders.
+  auto order = [](glow::runtime::RuntimeSymbolInfo i) {
+    if (i.input) {
+      if (!i.output) {
+        // input only
+        return 0;
```

Comment:
May be define an enum for these four cases? It would look cleaner. And add an explicit return type to the capture.

Reply:
> May be define an enum for these four cases? It would look cleaner. And add an explicit return type to the capture.

OK. I've defined a new enum PlaceholderType to make it clear